Src/AutoMoq/MockType.cs

```diff
@@ -68,11 +68,14 @@ internal static Type GetMockedType(this Type type)
             where TMock : class
         {
             return setup.Returns(() =>
-                {
-                    var result = (TResult) context.Resolve(typeof (TResult));
-                    setup.Returns(result);
-                    return result;
-                });
+            {
+                var specimen = context.Resolve(typeof (TResult));
+                TResult result = specimen is TResult ? (TResult) specimen : default(TResult);

```

Comment:
In my opinion, an `InvalidCastException` usually indicates a bug. On the other hand, an `ObjectCreationException` with a meaningful message (e.g., "Tried to create a string, found an OmitSpecimen" instead") will make it clear that there is no bug, and that the operation failed because `OmitSpecimen` is not supported.